### PR TITLE
[ENH] Add Cortex ScoreboardModel, updates to PolimeniMap

### LINKED ISSUE #514: [ENH] Add ScoreboardCortex
Add ScoreboardCortex, a scoreboard model implementation with support for multiple visual regions.

This should be refactored from https://github.com/bionicvisionlab/p2pcortex/blob/master/p2pcortex/models.py#L15

Will require the Polimeni retinotopy to be completed.

### PR DESCRIPTION
**Pull request by jgranley**
## Description
Adds the scoreboard model for cortex, as well as a number of other changes to Visual Field Maps, and `BaseModel`. This PR has some somewhat hacky solutions, so some items might need further discussion/revision

Changes:

1. Add `p2p.models.cortex.ScoreboardModel  `

    It works by calculating a scoreboard current spread for each simulated region (e.g. v1), and summing the results. This is basically an abstraction of the current scoreboard model class (you can even hack it to be the same by setting `model.regions = ['ret']`). It calls the cython code for the existing scoreboard model.

    Right now the naming is the same as the original, but it is located in `models.cortex` instead of `models`. If theres a better name we could change this. It also might be worth discussing if its even worth having this as a separate model, as opposed to just making the normal scoreboard model more general. I tend to think that it is worth it though, just because having a concrete separate cortical scoreboard model seems good.

2. Various changes to `Polimeni2006Map`
    - Add `jitter_boundary` parameter, defaulting to true. The polimeni map is undefined at dva x = 0, and, for v2 and v3, when y=0. Normal `xrange/yrange/xystep` parameters will almost always have values at these points. However, they are all currently set to NAN and not simulated, which is a waste, and causes percepts to look weird. You can force odd grid spacing (e.g. `xrange=(-14.999, 15.001)`, but the the output percepts are not very clean looking because the grid ticks are in weird places. Thus, it seemed more elegant to just add a tiny constant to the grid values that are on the boundary.
    - Add NAN's where theta >**=**  $\pi$/2, or theta <**=** - $\pi$/2 (include equality)
    - Make `dva_to_*` take in cartesian coordinates, and `*_to_dva `output cartesian coordinates
    - Add checks enforcing input to be np array
    - Removed nans for points across x=0 (see change 5)

3. Make `VisualFieldMap` inherit from `BaseModel`
    This is very useful for getting frozen, pretty print, and default_params functionality (doesn't use build). However it introduces a nasty circular import error. With this change, the module dependencies are: `topography` (`base.VisualFieldMap`)  ----(depends on)--->  `models` (`base.BaseModel`) --> `implants`, `percepts`, and `topography` (`ProsthesisSystem`, `Percept`, `Grid2D`); and additionally `implants` and `percepts` modules both depend on `topography` (`Grid2D`). 

    There are a few options to resolve this, and am definitely open to discussion on which is best:
    - **a) Import dependencies at runtime (current solution).** `VisualFieldMap` cannot do this because it inherits from `BaseModel`. Instead, `models.base` now imports from `topography` at runtime. In addition, `implants` and `percepts` now also import `Grid2D` at runtime. This is somewhat inelegant, but it works. Notably, this will not affect the user, who can import anything in any order they want.
    - **b) Move `BaseModel` into utils.** This could also make sense, because multiple things (spatial/temporal models, biphasic effects models, and Visual Field Maps) all use `BaseModel`. I originally implemented this, but it just seemed wrong to have the base model class not be in `models.base`.
    - **c) Move `Grid2D` into `models.base`.** This also could make some sense, since pretty much the only things that use `Grid2D` are models. But conceptually `Grid2D` is very much a topographical objects, so staying in the topography module seems better

4. Change Grid2d.plot to look nice with discontinuous visual areas.
    This is super hacky right now, but basically it checks the name of the transform to see if one of the discontinuous regions is in the name. If so, it alters plotting behavior so the boundaries don't mess up cells/hulls. Shouldnt be hacky anymore after #512 

5. Modified how visual field maps handle both hemispheres
    TLDR: The coordinate system specified by CorticalMap has the right hemisphere fovea at (0,0), and left hemisphere fovea at (left_offset, 0) to allow simulating both hemispheres on the same cortical sheet. The model enforces that no current can flow across the hemispheres.

    Before, there was a check in `add_nans` in Polimeni that would remove any points with cortical x coordinate < 0. This was based on the assumption that the fovea was at x=0 and that you couldn't go left of the fovea in the right hemisphere. This allowed us to put both hemispheres on separate halves of the same coordinate plane.

    However.... I have since learned that the wedge dipole mapping DOES cross the boundary, and you are supposed to used separate maps for each hemisphere. Thus, we were discarding many valid points in v3 and v2.

    Implementing a separate coordinate grid for each hemisphere is probably the best solutions... but this is cumbersome, and does at all fit into the existing p2p model flow. Instead, I kept it so we do both hemispheres on one set of coordinates.  Now, `CorticalMap` is modified so that the left hemisphere fovea is always located at x = -`left_offset` (set to 20000 by default). I also modified the fast scoreboard cython so that it explicitly disallows current to spread from the right to the left hemisphere. 

    Example showing the fovea of left hemisphere is no longer at (0, 0):
image

7. A few random doc updates

Closes #514 

## Type of Change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change

### CODE REVIEW

**Comment by jgranley** [comment]
Basically finished, but some tests are failing only on mac?? Hard to reproduce
